nixos/modules/services/misc/sonarr.nix

```diff
@@ -52,11 +152,20 @@ in
       description = "Sonarr";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-
+      environment = settingsEnvVars;
       serviceConfig = {
         Type = "simple";
         User = cfg.user;
         Group = cfg.group;
+        EnvironmentFile = cfg.environmentFiles;
+        LoadCredential = lib.optional (cfg.apiKeyFile != null) "SONARR__AUTH__APIKEY:${cfg.apiKeyFile}";
+        ExecStartPre = lib.optionalString (cfg.apiKeyFile != null) pkgs.writeShellScript "sonarr-apikey" ''
```

Comment:
Shouldn't this also be `lib.optional`? `ExecStartPre = ""` will unset things.

Reply:
@K900 thanks for reverting. I a a bit surprised why this happened.

Reply:
Order of operations.